### ISSUE TITLE
[Bug]: Wrong Resolution for `?` and `+` in Accelerators on Windows on non-en layouts

### ISSUE THREAD

**Issue by nathanlesage** [electron/electron#32259]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

15.3.4

### What operating system are you using?

Windows

### Operating System Version

Windows 10

### What arch are you using?

x64

### Expected Behavior

As of [this comment by Cheng Zhao](https://github.com/electron/electron/pull/29202/files#r643582917), assigning `Ctrl+?` to a MenuItem Accelerator should result in having to press `Ctrl+Shift+-` on German keyboards. Furthermore, assigning `Ctrl+Plus` should result for German users having to press `Ctrl+Shift+´`.

### Actual Behavior

On German keyboard layouts, assigning `Ctrl+?` as an accelerator necessitates having to press `Ctrl+Shift+#` (equivalent of `Ctrl+'`) which seems very arbitrary and corresponds to the location of the Backslash/Pipe character on a US-keyboard (instead of where the question mark actually is).

Similarly, `Ctrl+Plus` is resolved wrongly. On a German keyboard, one has to press `Ctrl+Asterisk`, which is equivalent to `Ctrl+Shift+Plus`. If the accelerators would be translated into the corresponding ANSI-key positions and then displayed correctly, a German user should actually need to press `Ctrl+Shift+´` (equivalent: `Ctrl+Backtick`).

### Additional Information

I know this is a long-standing issue, but it is excruciatingly frustrating. Plus, the corresponding documentation entry on the Electron website does neither explain that the internal representation is focused on physical locations on en-US keyboards, nor does it explain cases in which that goes wrong, since the translation apparently does not work as expected (if said comment is correct).

In short: I have no idea what key Windows users will need to press when I assign some shortcut for my menu and it appears to not follow any predictable rule. That only works for macOS.
